src/main/java/com/microsoft/azure/vmagent/AzureVMAgentTemplate.java

```diff
@@ -434,6 +441,7 @@ public AzureVMAgentTemplate(
             String agentWorkspace,
             String jvmOptions,
             RetentionStrategy<?> retentionStrategy,
+            List<? extends NodeProperty<?>> nodeProperties,
```

Comment:
The constructor should only have new parameters added for mandatory properties, optional values should be added with a setter annotated with `DataBounderSetter` and a getter.

You'll be able to revert the changes to the config.xml then as its a migration testing file